hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotLocalData.java

```diff
@@ -55,17 +56,17 @@ public abstract class OmSnapshotLocalData {
 
   // Map of version to compacted SST file list
   // Map<version, Map<Table, sstFileList>>
-  private Map<Integer, Map<String, List<String>>> compactedSSTFileList;
+  private Map<Integer, Map<String, Set<String>>> compactedSSTFileList;
 
   public static final Charset CHARSET_ENCODING = StandardCharsets.UTF_8;
   private static final String DUMMY_CHECKSUM = new String(new byte[64], CHARSET_ENCODING);
 
   /**
    * Creates a OmSnapshotLocalData object with default values.
    */
-  public OmSnapshotLocalData() {
+  public OmSnapshotLocalData(Map<String, Set<String>> uncompactedSSTFileList) {
     this.isSSTFiltered = false;
-    this.uncompactedSSTFileList = new HashMap<>();
+    this.uncompactedSSTFileList = uncompactedSSTFileList;
```

Comment:
Potential NPE: https://github.com/jojochuang/ozone/pull/604#discussion_r2220568417
```suggestion
    this.uncompactedSSTFileList = uncompactedSSTFileList != null ? uncompactedSSTFileList : new HashMap<>();
```